src/order/hom/basic.lean

```diff
@@ -683,3 +683,22 @@ theorem order_iso.is_complemented_iff :
 
 end bounded_order
 end lattice_isos
+
+section boolean_algebra
+variables (α) [boolean_algebra.core α]
+
+/-- Taking complements as an order isomorphism to the order dual. -/
+def order_iso.compl : α ≃o order_dual α :=
+{ to_fun := compl,
+  inv_fun := (compl : α → α),
```

Comment:
The type-correct definition, which you definitely need if you go for the `@[simps]` solution.
```suggestion
{ to_fun := to_dual ∘ compl,
  inv_fun := compl ∘ of_dual,
```